test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -739,6 +740,11 @@ public void testSupportedFieldTypes() throws IOException {
         }
     }
 
+    private ValuesSourceType fieldToVST(MappedFieldType fieldType) {
+        return fieldType.fielddataBuilder("")
+                                .build(createIndexSettings(), fieldType, null, null, null).getValuesSourceType();
+    }
```

Comment:
Something about this method smells funky to me. I can see why we have it though. I'd sort of hoped for a cleaner way to get this, but I understand. Might be worth poking at in a follow up.

Reply:
Yeah, I don't like this method either.  Personally, I think the smell is from those nulls.  Seems like something we shouldn't be passing nulls into, since the resulting object will be kind of nonsense.  But we just need to get the VST off of it here, so it's okay-ish.  Honestly, I was shocked that passing in nulls didn't throw, but here we are.

I spent almost two days trying to come up with a better idea, but didn't get there.  What seems right to me is to have one instance of the mock QueryShardContext for the entirety of a test run, then we can just get the IndexFieldData via the QueryShardContext like we do in production.  But when I tried to do that, I ran into all sorts of trouble.  It ends up needing a lot of mocks to be reused, and there's a lot of places in the tests where we implicitly assume we're getting fresh mocks.  I think it's fixable, but I didn't want to block this change on cleaning up our tests.  Especially since we've been talking about refactoring the tests anyway.